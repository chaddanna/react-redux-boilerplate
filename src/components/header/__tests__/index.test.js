import { mount, shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Component from '..';

describe('Component', () => {
  let props;

  beforeEach(() => {
    props = {
      title: ''
    };
  });

  describe('Smoke tests', () => {
    it('snapshot test', () => {
      expect(
        renderer.create(<Component { ...props } />).toJSON()
      ).toMatchSnapshot();
    });

    it('renders without crashing', () => {
      shallow(<Component { ...props } />);
    });
  });

  describe('Mounted tests', () => {
    let mountedComponent;
    const getComponent = () => {
      if (!mountedComponent) {
        mountedComponent = mount(<Component { ...props } />).find(Component);
      }

      return mountedComponent;
    };

    beforeEach(() => {
      mountedComponent = undefined;
    });

    it("always renders 'header'", () => {
      expect(getComponent().find('.header')).toHaveLength(1);
    });

    describe("rendered 'header' with title set", () => {
      beforeEach(() => {
        props.title = 'Test';
      });

      it('sets header with Test', () => {
        expect(
          getComponent()
            .find('.header')
            .text()
        ).toBe(`${props.title}`);
      });
    });
  });
});
