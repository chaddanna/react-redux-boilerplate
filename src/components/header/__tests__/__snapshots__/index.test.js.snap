// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Smoke tests snapshot test 1`] = `
<header
  className="header"
>
  <img
    alt="logo"
    className="logo"
    src="logo.svg"
  />
  <h1
    className="title"
  >
    
  </h1>
</header>
`;
